programs/loader-v4/src/lib.rs

```diff
@@ -200,10 +200,6 @@ fn process_instruction_truncate(
             ic_logger_msg!(log_collector, "Program is not writeable");
             return Err(InstructionError::InvalidArgument);
         }
-        if !program.is_signer() {
```

Comment:
One has to assign loader-v4 as owner of the program account using the system program, which does check the program account being the signer.